Sources/SnapshottingSwift/Snapshots.swift

```diff
@@ -70,30 +84,35 @@ class Snapshots {
 
   private func generateSnapshot() {
     guard !previews.isEmpty else {
-      completion?()
+      writeResults()
       return
     }
 
     let (preview, typeName) = previews.removeFirst()
-    var view = preview.view()
-    if let colorScheme = preview.colorScheme() {
-      view = AnyView(view.colorScheme(colorScheme))
-    }
-    let fileName = fileName(typeName: typeName, preview: preview)
-    let file = Self.resultsDir.appending(path: fileName, directoryHint: .notDirectory)
-    print(file)
-    view.snapshot(layout: preview.layout, window: window, async: false) { image in
-          if let pngData = image.pngData() {
-              do {
-                  try pngData.write(to: file)
-              } catch {
-                  print("Failed to write \(file)")
-                  print(error)
-              }
-          } else {
-              print("Could not generate PNG data for \(file)")
-          }
-      self.generateSnapshot()
+    do {
+      var view = try preview.view()
+      if let colorScheme = try preview.colorScheme() {
+        view = AnyView(view.colorScheme(colorScheme))
+      }
+      let fileName = fileName(typeName: typeName, preview: preview)
+      let file = Self.resultsDir.appending(path: fileName, directoryHint: .notDirectory)
+      print(file)
+      view.snapshot(layout: preview.layout, window: window, async: false) { image in
+            if let pngData = image.pngData() {
+                do {
+                    try pngData.write(to: file)
+                } catch {
+                    print("Failed to write \(file)")
```

Comment:
Should this case also be included in the errors?

Reply:
Good idea